src/plugins/workspace/public/components/workspace_form/workspace_collaborator_table.tsx

```diff
@@ -150,26 +162,56 @@ export const WorkspaceCollaboratorTable = ({
     );
   }, [displayedCollaboratorTypes, permissionSettings, handleSubmitPermissionSettings]);
 
-  const openDeleteConfirmModal = ({ onConfirm }: { onConfirm: () => void }) => {
+  const openDeleteConfirmModal = ({
+    onConfirm,
+    selections,
+  }: {
+    onConfirm: () => void;
+    selections: PermissionSettingWithAccessLevelAndDisplayedType[];
+  }) => {
+    const adminOfSelection = selections.filter(
+      (item) => item.accessLevel === WORKSPACE_ACCESS_LEVEL_NAMES.admin
+    ).length;
     const modal = overlays.openModal(
-      <EuiConfirmModal
-        title={i18n.translate('workspace.detail.collaborator.actions.delete', {
-          defaultMessage: 'Delete collaborator',
-        })}
-        onCancel={() => modal.close()}
-        onConfirm={onConfirm}
-        cancelButtonText="Cancel"
-        confirmButtonText="Confirm"
-      >
-        <EuiText>
-          <p>
-            {i18n.translate('workspace.detail.collaborator.delete.confirm', {
-              defaultMessage:
-                'Delete collaborator? The collaborators will not have access to the workspace.',
-            })}
-          </p>
-        </EuiText>
-      </EuiConfirmModal>
+      adminCollarboratorsNum === adminOfSelection && adminCollarboratorsNum !== 0 ? (
+        <EuiConfirmModal
```

Comment:
It seems the only difference is the modal content, can we just do a if else when rendering the modal content instead of in this very beginning place so that we can reuse most of the code.